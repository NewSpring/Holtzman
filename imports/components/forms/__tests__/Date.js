import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import DayPicker from "react-day-picker";
import mockDate from "mockdate";
import DateComponent from "../Date";

jest.mock("react-day-picker", () => jest.fn());

const defaultProps = {
  format: jest.fn(),
  defaultValue: "test",
  today: true,
  past: true,
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <DateComponent { ...newProps } />;
};

// XXX god bless you
mockDate.set("1/1/2000");

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders date picker", () => {
  const wrapper = shallow(generateComponent());
  wrapper.setState({ showDatePicker: true });
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("onDayClick returns if disabled", () => {
  const wrapper = shallow(generateComponent());
  wrapper.instance().onDayClick({}, 23, { selected: true, disabled: true });
  expect(wrapper.state().selectedDay).toBe(null);
});

it("onDayClick updates state if not disabled or selected", () => {
  const wrapper = shallow(generateComponent());
  const date = new Date();
  wrapper.instance().onDayClick({}, date, { selected: false, disabled: false });
  expect(wrapper.state().selectedDay).toBe(date);
});

it("onDayClick updates state to null if selected", () => {
  const wrapper = shallow(generateComponent());
  const date = new Date();
  wrapper.instance().onDayClick({}, date, { selected: true, disabled: false });
  expect(wrapper.state().selectedDay).toBe(null);
});

it("toggle updates state", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.state().showDatePicker).toBe(false);
  wrapper.instance().toggle();
  expect(wrapper.state().showDatePicker).toBe(true);
  wrapper.instance().toggle();
  expect(wrapper.state().showDatePicker).toBe(false);
});
