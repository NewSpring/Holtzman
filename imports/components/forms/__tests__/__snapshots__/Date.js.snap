exports[`test renders date picker 1`] = `
<div className="display-inline-block" style={{...}}>
<Styles />
<div style={{...}}>
<Input format={[Function]} defaultValue={[undefined]} today={true} past={true} />
<div className="locked-ends locked-sides" onClick={[Function]} />
</div>
<div id="datepicker">
<div style={{...}}>
<mockConstructor locale="en" localeUtils={{...}} modifiers={{...}} onDayClick={[Function]} />
<div className="background--light-secondary soft text-center">
<button className="btn flush btn--small" onClick={[Function]}>
Done
</button>
</div>
</div>
<div style={{...}} onClick={[Function]} />
</div>
</div>
`;

exports[`test renders with props 1`] = `
<div className="display-inline-block" style={{...}}>
<Styles />
<div style={{...}}>
<Input format={[Function]} defaultValue={[undefined]} today={true} past={true} />
<div className="locked-ends locked-sides" onClick={[Function]} />
</div>
</div>
`;
